### PR TITLE
feat: upload example works with big files

### PR DESCRIPTION
**Pull request by lidel**
Old example used deprecated practice of  pre-allocating a Buffer from the file handle.
While it worked fine for small files, with bigger ones introduced artificial limit of ~2GB and caused `RangeError: Attempt to allocate Buffer larger than maximum size: 0x7fffffff bytes` 

People look at  `examples/upload-file-via-browser` to learn "best practices", so I updated it to follow what we learned and do in ipfs-companion, ipfs-webui and ipfs-share-files.

**tl;dr**

- This change replaces Buffer with a pull stream, which enables example to work with big files 
(tested with 3.6GB video, took ~30seconds to add to local go-ipfs).

- While I was at it, I added a demo of a pattern for persisting filenames by wrapping them in unixfs directory (very popular question). 

#### Additional Resources
- https://github.com/ipfs/in-web-browsers/issues/115: Improving Uploads of Big Files via Web Browsers

### CODE REVIEW

**Comment by diasdavid** [comment]
CI failed with no-space left on device, as for the rest, looks super cool!

**Comment by rori4** [comment]
I tested the example and Chromium crashed as the file was too big. The file that I tried to upload was around 400MB

**Comment by lidel** [comment]
@rori4 please open [a new issue](https://github.com/ipfs/js-ipfs-http-client/issues/new), including the version of your Chromium. Thank you!
